### PR TITLE
Feature/marketing csv

### PR DESCRIPTION
**Pull request by jace-garcia**
For downloading a csv containing Badgr users who opted in to marketing. Includes a date query parameter. Note implemented for v1 api.
The new url for this feature is /v1/user/opted-marketing?date=DATE where DATE is of the form YEAR-MONTH-DAY
ex) /v1/user/opted-marketing?date=2018-05-03

### CODE REVIEW

**Comment by ottonomy** [comment]
Please use class-based views like the other endpoints in that file and make sure that only staff users may request the report.

File: apps/issuer/tasks.py [hunk 1]
```diff
@@ -83,3 +85,35 @@ def rebake_assertion_image(self, assertion_entity_id=None, obi_version=CURRENT_O
...
         'success': True
     }
+
+@app.task(bind=True)
+def correct_issued_on_imported_assertions(self):
+    # get imported assertions
+    assertions = BadgeInstance.objects.filter(source_url__isnull=False)
+    updated_assertions = []
+    for a in assertions:
+        result = compare_issued_and_created_dates_for_assertion(a)
```
Comment (coffindragger) [inline apps/issuer/tasks.py]: this is calling the function directly and not queueing a job.
you need to use .delay() or .apply_async() on the task function to queue a job and not call the function immediately

File: apps/issuer/helpers.py [hunk 2]
```diff
@@ -192,9 +192,22 @@ def get_or_create_assertion(cls, url=None, imagefile=None, assertion=None, creat
             badgeclass, badgeclass_created = BadgeClass.objects.get_or_create_from_ob2(issuer, badgeclass_obo, original_json=original_json.get(badgeclass_obo.get('id')))
             return BadgeInstance.objects.get_or_create_from_ob2(badgeclass, assertion_obo, recipient_identifier=recipient_identifier, original_json=original_json.get(assertion_obo.get('id')))
 
+    @classmethod
```
Comment (coffindragger) [inline apps/issuer/helpers.py]: after we originally discussed this, i realized we actually already have this json stored in the database as BadgeInstance.original_json.

we should refactor this task to check .orignal_json for the 'issuedOn' date instead and then we wont need to make another HTTP call to refetch the json

File: apps/badgeuser/api.py [hunk 3]
```diff
@@ -415,5 +415,3 @@ def get(self, request, **kwargs):
     )
     def delete(self, request, **kwargs):
         return super(AccessTokenDetail, self).delete(request, **kwargs)
-
```
Comment (coffindragger) [inline apps/badgeuser/api.py]: please try to avoid unnecessary whitespace changes.

files should end with a newline.

File: apps/issuer/tasks.py (hunk 1 above)
Comment (jace-garcia) [reply apps/issuer/tasks.py]: updated on feature/imported_assertions_issuedOn_fix branch
